### DISCUSSION TITLE
Allow custom color on copilot suggestions.

### DISCUSSION THREAD

**Discussion by ferny1313** [community/community#7447] [Copilot Conversations] [answered]
It would be nice to customize the color of copilot suggestions. Sometimes I think the suggestions are commented code for the color.

**Comment by jansorg** [comment]
Thanks! Which editor are you using?

**Comment by Krzysztof-Cieslak** [comment, marked as answer]
In VSCode you can change the styling of the suggestions with the following entry in the settings.json file:
```
"workbench.colorCustomizations": {
        "editorGhostText.border": "#ff0000",
        "editorGhostText.foreground": "#ff0000",
}
```

**Reply by martin-marko** [reply]
What about PyCharm?

**Reply by jansorg** [reply]
@martin-marko The latest update for JetBrains IDEs (1.1.0) now allows to define your own color in the settings.

**Reply by mathandy** [reply]
@jansorg where is this option?  I see "completion" under "Popups and Hints" but that doesn't affect the in-line code completions.

Also just to add some explanation to why font color is bothering me a bit here:
When using Darkula, the in-line code completions come out as almost the exact same color as "unused symbol" which gets pretty confusing in practice.   E.g. I typed "hidden_units=(512, 512)," the rest of the gray is copilot:
![image](https://user-images.githubusercontent.com/3175485/159144490-a1d1b83c-e6ca-4174-8ca8-2f54058d6751.png)

I'd change "unused symbol", but, gray just seems like a good color for that.

**Reply by qwinsi** [reply]
> @jansorg where is this option? I see "completion" under "Popups and Hints" but that doesn't affect the in-line code completions.
> 
> Also just to add some explanation to why font color is bothering me a bit here: When using Darkula, the in-line code completions come out as almost the exact same color as "unused symbol" which gets pretty confusing in practice. E.g. I typed "hidden_units=(512, 512)," the rest of the gray is copilot: ![image](https://user-images.githubusercontent.com/3175485/159144490-a1d1b83c-e6ca-4174-8ca8-2f54058d6751.png)
> 
> I'd change "unused symbol", but, gray just seems like a good color for that.

The settings of Copilot had already been added into JetBrains IDE's settings panel.
On my machine(PyCharm on Linux), I could find it by: 
`File` -> `Settings` -> `Languages & Frameworks` -> `Github Copilot`

Or you can just type "copilot" into the search bar of IDE's `Settings` panel to find it.

**Reply by mathandy** [reply]
Perfect!  Thanks @qwinsi 
Works for me on my mac.
`Preferences` -> `Languages & Frameworks` -> `GitHub Copilot`
![image](https://user-images.githubusercontent.com/3175485/161320068-5536211c-cf50-4cbc-8649-36a1b9d6dc1a.png)

**Reply by huyz** [reply]
It would be nice if these colors were Copilot-specific. I keep switching between different tools that autocomplete, and I'd like to know if suggestions come from Copilot.  It seems that IntelliJ it's possible, but not in VS Code.

**Reply by plaa** [reply]
In a recent (4/2025) update Copilot suggestions are now syntax-highlighted.  This annoys me a lot, as it visually mixes up what's a suggestion and what's actual code, despite the suggestion being in italics.  The `editorGhostText.border` setting still works but `editorGhostText.foreground` has no effect.  Anyone know any way to get the suggestions back to gray?

**Reply by coredumperror** [reply]
Seconded! Suggestions being syntax-highlighted is awful!

**Reply by plaa** [reply]
I made a feature request of this, feel free to react to it:  https://github.com/microsoft/vscode-copilot-release/issues/8722

**Reply by steinhh** [reply]
That one was closed saying "there's an option for that" but without specifying the option! So I created another issue, upvote that too: https://github.com/microsoft/vscode-copilot-release/issues/8957 @huyz @mathandy 

It's even more annoying now that I know there's a solution, we're just not told what it is X-D

**Reply by plaa** [reply]
Copying the solution here as well:

To disable syntax highlighting:
```
"editor.inlineSuggest.syntaxHighlightingEnabled": false
```

To change the suggestion color after that:
```
"workbench.colorCustomizations": {
  "editorGhostText.foreground": "#YOUR_COLOR"
}
```
or the font:
```
"editor.inlineSuggest.fontFamily": "'Font Name'",
```

**Reply by parvezhossainme** [reply]
Man you saved me ! Thanks a lot!

**Comment by domx4q** [comment]
Now the feauture was added tp PyCharm under `Languages & Framework > Github Copilot` but it's not working (even after restartign the IDE). It has now always the color of default text and it's annoying. Does anyone have an idea how to change it?

**Comment by WhiteWolfDD** [comment]
I have new issue. After updating Github Copilot now I have white color for suggestion, not gray what before.

**Reply by pouyarezvani** [reply]
@WhiteWolfDD  Same actually. And it's not changing. This seems to be a bug

**Reply by yulinJoseph** [reply]
I got the same issue. Have you solved the problem?

**Reply by imxintian** [reply]
me，too

**Reply by markwillis** [reply]
I see the updated default color for the CoPilot suggestion briefly (500ms or so), then immediately overridden back to white. Super annoying.

**Reply by mmelikyan** [reply]
see workaround here: https://github.com/orgs/community/discussions/28915#discussioncomment-3369030

**Reply by pawel-slusarczyk** [reply]
For people that find this thread when searching for Copilot customisation in 2024+. The setting from the linked workaround is not affecting CoPilot anymore. Instead follow this suggestion: https://github.com/orgs/community/discussions/7447#discussioncomment-2487798

**Comment by martin-marko** [comment]
New update ruined the colors, now it's either the default text color or something completely random, I've seen everything from red to green.

**Reply by RafailFridman** [reply]
Yes, my suggestion colors vary between white, yellow and red

**Comment by domx4q** [comment]
Now on the newest nightly build fixes the issue and everything (for the colors) works as expected.
Version: `1.1.29.1881-nightly`

**Reply by avarayr** [reply]
I'll probably stay downgraded until they release a stable version.

**Comment by SimonDedman** [comment]
In RStudio IDE, one can customise the colour themes if the existing selection of options are lacking (info [here](https://stackoverflow.com/questions/25582588/any-way-to-change-colors-in-rstudio-to-something-other-than-default-options)). Does anyone know what the ACE element name(s) are for Copilot? For reference it'll likely look something like this:

```
.ace_marker-layer .ace_selected-word {
  background: #141414;
  border: 1px solid #FF0000;
}
```

**Comment by FaltuG** [comment]
Did anyone got the solution for the copilot completions color change .
I want to make it like the old one grey color 
Right now it's matching with the theme which I have selected in vscode 
Plzz tell me some solution as it makes soo much confusions when the suggestion and text written seems same there 😭😭

**Reply by plaa** [reply]
Yeah, it was finally answered [here](https://github.com/microsoft/vscode-copilot-release/issues/8957#issuecomment-2849153376).

To disable syntax highlighting:
```
"editor.inlineSuggest.syntaxHighlightingEnabled": false
```

To change the suggestion color after that:
```
"workbench.colorCustomizations": {
  "editorGhostText.foreground": "#YOUR_COLOR"
}
```
or the font:
```
"editor.inlineSuggest.fontFamily": "'Font Name'",
```

**Reply by FaltuG** [reply]
Thanks bro for helping
